### ISSUE TITLE
Ranksystem deletes many users

### ISSUE THREAD

**Issue by LordRofl** [Newcomer1989/TSN-Ranksystem#123]
The Ranksystem deletes many users in the last days, can this be a mistake?

2016-10-01 23:21:06.495641   NOTICE      Es wurden 9 Clients aus der Ranksystem-Datenbank gelöscht, da sie nicht mehr in der TeamSpeak Datenbank vorhanden sind.
2016-10-01 23:21:06.494351   NOTICE      Clients in der Ranksystem Datenbank: 150
2016-10-01 23:21:06.494184   NOTICE      Clients in der TeamSpeak Datenbank gefunden: 141 (von 169)
2016-10-01 23:21:06.443849   NOTICE    Scanne nach Usern, welche zu löschen sind...

9 Users at once? Never happend before but now its very often that this happens.

**Comment by Shad86** [comment]
As it stand there

`2016-10-01 23:21:06.495641 NOTICE Es wurden 9 Clients aus der Ranksystem-Datenbank gelöscht, da sie nicht mehr in der TeamSpeak Datenbank vorhanden sind.`

Seems someone cleaned up your TS3 Database and the ranksystem cleans out now the users.

https://puu.sh/rA38J/ace844f289.png

click on (red marked) "Clientlöschung" and "Löschintervall" then you get an explanation what it does.

In short words -> if the user doesn't exist in the ts3-server database anymore the ts3 ranksystem deletes it from the tsn database

regards

Shad86

**Comment by Newcomer1989** [comment]
The Ranksystem only synchronize the TS3 database with the Ranksystem database. That means user, which are deleted on the TS3 server, will also be deleted out of the Ranksystem database.

The shown interval means only the interval how often the Ranksystem check "there are something to delete". This is **not** the interval after how much time user gets deleted from the TS3 server. The Ranksystem do not touch (delete) any user on your TS3 server!

To change the time after which user should be deleted out of the TS3 server you have to edit the ts3server.ini (should be found in the root directory of your TS3 server installation).
There you will find the option
`dbclientkeepdays=30`

If I remember it right the default value are 30 days and TeamSpeak ignore all values under 30.

**Comment by LordRofl** [comment]
Jea it was the `dbclientkeepdays=30` option.

We reinstalled the masterfiles and forgot about the previous change with that setting. Fuck sake! 

Thank you! 👍

**Comment by bykidi** [comment]
Should be closed imo.

**Comment by Newcomer1989** [comment]
I want to create with this topic an FAQ entry. Thats why it is still open (as reminder).

If the FAQ entry is done, it will be closed. The same is with the other opened "questions".

**Comment by Newcomer1989** [comment]
FAQ entry done.. so closed
